lab_3_generate_by_ngrams/main.py

```diff
@@ -332,6 +514,33 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not isinstance(sequence, tuple) or not sequence:
+            return None
+        if not check_list(next_tokens, tuple, False):
+            return None
+        if len(next_tokens) > self._beam_width:
+            return None
+        for item in next_tokens:
+            if not isinstance(item, tuple) or len(item) != 2:
+                return None
+            if not isinstance(item[0], int) or not isinstance(item[1], float):
+                return None
+        if not check_dict(sequence_candidates, tuple, float, True):
+            return None
+        if sequence not in sequence_candidates:
```

Comment:
13 lines for checks
13 lines for code

Can you reuse some functions for typechecking?